c7n/resources/vpc.py

```diff
@@ -1220,33 +1220,53 @@ def process(self, resources):
             client.delete_security_group(GroupId=r['GroupId'])
 
 
-@SecurityGroup.action_registry.register('remove-permissions')
-class RemovePermissions(BaseAction):
-    """Action to remove ingress/egress rule(s) from a security group
+@SecurityGroup.action_registry.register('set-permissions')
+class SetPermissions(BaseAction):
+    """Action to update ingress/egress rule(s) of a security group
 
     :example:
 
     .. code-block:: yaml
 
             policies:
-              - name: security-group-revoke-8080
+              - name: modify-security-group
                 resource: security-group
                 filters:
-                  - type: ingress
-                    IpProtocol: tcp
-                    Ports: [8080]
+                  - or:
+                        - type: ingress
+                          IpProtocol: "-1"
+                          Ports: [22, 3389]
+                          Cidr: "0.0.0.0/0"
+                        - type: ingress
+                          IpProtocol: "-1"
+                          Ports: [22, 3389]
+                          CidrV6: "::/0"
                 actions:
-                  - type: remove-permissions
+                  - type: set-permissions
                     ingress: matched
+                    PermissionType: "ingress"
+                    IpProtocol: "TCP"
+                    FromPort: 22
+                    ToPort: 22
+                    Cidr: ["1.1.1.1/32","2.2.2.2/32","3.3.3.3./32"]
 
     """
     schema = type_schema(
-        'remove-permissions',
+        'set-permissions',
         ingress={'type': 'string', 'enum': ['matched', 'all']},
-        egress={'type': 'string', 'enum': ['matched', 'all']})
+        egress={'type': 'string', 'enum': ['matched', 'all']},
+        PermissionType={'type': 'string', 'enum': ['ingress', 'egress']},
```

Comment:
i think we need a clearer syntax here, it feels like we're mixing in add and remove attributes at the top level of the action without really distinguishing which is which.

what about this style as alternative?

```yaml
type: set-permissions
remove-ingress: matched
remove-egress: matched
add-ingress: [array of permissions dictionaries]
add-egress: [array of permissions dictionaries]
```
note its also enabling setting multiple permissions.